File: app/src/main/java/com/intkhabahmed/smartnotes/viewmodels/SearchNotesViewModelFactory.java
package com.intkhabahmed.smartnotes.viewmodels;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class SearchNotesViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private String title;
    private int trashed;

    public SearchNotesViewModelFactory(@Nullable String title, int trashed) {
        this.title = title;
        this.trashed = trashed;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new SearchNotesViewModel(title, trashed);
    }
}

File: app/src/main/java/com/intkhabahmed/smartnotes/viewmodels/SearchNotesViewModelFactory.kt
package com.intkhabahmed.smartnotes.viewmodels

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory

class SearchNotesViewModelFactory(private val title: String?, private val trashed: Int) : NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(NoteViewModel::class.java)) {
            return SearchNotesViewModel(title, trashed) as T
        }
        throw ClassCastException("Unknown ViewModel Class")
    }

}